test/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersPartnershipConfirmRegisteredAddressControllerSpec.scala

```diff
@@ -33,88 +42,127 @@ class BusinessPartnersPartnershipConfirmRegisteredAddressControllerSpec
+  val registeredAddressPage: String =
+    routes.BusinessPartnersPartnershipRegisteredAddressController.load(1, NormalMode).url
+  val checkYourAnswersPage = routes.BusinessPartnersCheckYourAnswersController.load().url
+
+  def createUserAnswers(answers: UkAddressLookup): UserAnswers =
+    UserAnswers(testUserId)
+      .set[UkAddressLookup](UkAddressLookupPage(1), answers)
+      .success
+      .value
 
   "load" should {
     "Render the confirm address page" when {
-      "the new business partner pages are enabled" in {
-        setupUserAction()
-        when(mockAppConfig.newBusinessPartnerPagesEnabled).thenReturn(true)
+      "a single address is found in user answers" in {
+        val userAnswers = createUserAnswers(singleUkAddressLookup)
+        setupDataRequiredAction(userAnswers, NormalMode)
+
         val request = FakeRequest()
-        val result = await(csrfAddToken(controller.load())(request))
+        val result = await(csrfAddToken(controller.load(1, NormalMode))(request))
 
         status(result) shouldBe OK
         val page = Jsoup.parse(contentAsString(result))
-        // should be mocked out when Save4Later changes included
         page.title() should include(pageTitle)
         page.getElementsByTag("h1").text should include("Confirm the company’s registered office address")
-        page.body.text should include("1 Romford Road")
-        page.getElementById("confirm-edit").attr("href") should include("#")
+        page.body.text should include("44 test lane")
+        page.getElementById("confirm-edit").attr("href") should include(enterAddressUrl)
         reset(mockActions)
       }
     }
 
-    "return 200" when {
-      "the form has no errors and the address is found" in {
-        setupUserAction()
-        when(mockAppConfig.newBusinessPartnerPagesEnabled).thenReturn(true)
+    "Redirect to Registered Address Page" when {
+      "address lookup address list is empty" in {
+        val userAnswers = createUserAnswers(emptyUkAddressLookup)
+        setupDataRequiredAction(userAnswers, NormalMode)
+
         val request = FakeRequest()
-        val result = await(csrfAddToken(controller.load())(request))
+        val result = await(csrfAddToken(controller.load(1, NormalMode))(request))
 
-        status(result) shouldBe OK
-        val page = Jsoup.parse(contentAsString(result))
-        page.title() should include(pageTitle)
+        status(result) shouldBe SEE_OTHER
+        redirectLocation(result) shouldBe Some(registeredAddressPage)
         reset(mockActions)
       }
-    }
 
-    "Render the page not found page" when {
-      "the new business partner pages are disabled" in {
-        setupUserAction()
-        when(mockAppConfig.newBusinessPartnerPagesEnabled).thenReturn(false)
+      "address lookup address list contains more than one address" in {
+        val userAnswers = createUserAnswers(emptyUkAddressLookup)
```

Comment:
`createUserAnswers` should take in `multipleAddressLookup` for this scenario - "address list contains more than one address".